auth/src/main/java/org/openedx/auth/presentation/signup/SignUpViewModel.kt

```diff
@@ -231,15 +272,22 @@ class SignUpViewModel(
     }
 
     fun updateField(key: String, value: String) {
-        _uiState.update {
-            val updatedFields = uiState.value.allFields.toMutableList().map { field ->
-                if (field.name == key) {
-                    field.copy(placeholder = value)
-                } else {
-                    field
-                }
+        val updatedFields = uiState.value.allFields.toMutableList().map { field ->
+            if (field.name == key) {
+                field.copy(placeholder = value)
+            } else {
+                field
+            }
+        }
+        updateFields(updatedFields)
+    }
+
+    fun openLink(fragmentManager: FragmentManager, links: Map<String, String>, link: String) {
```

Comment:
same here.

Reply:
this made to prevent writing logic on the UI level